OrbitClientGgp/ClientGgp.cpp

```diff
@@ -79,9 +78,107 @@ bool ClientGgp::StopCapture() {
   return capture_client_->StopCapture();
 }
 
-void ClientGgp::InitCapture() {
+std::shared_ptr<Process> ClientGgp::GetOrbitProcessByPid(int32_t pid) {
+  // We retrieve the information of the process to later get the module corresponding to its binary
+  ErrorMessageOr<std::vector<ProcessInfo>> result_process_infos = process_client_->GetProcessList();
+  if (result_process_infos.has_error()) {
+    return nullptr;
+  }
+  const std::vector<ProcessInfo>& process_infos = result_process_infos.value();
+  LOG("List of processes:");
+  for (const ProcessInfo& info : process_infos) {
+    LOG("pid:%d, name:%s, path:%s, is64:%d", info.pid(), info.name(), info.full_path(),
+        info.is_64_bit());
+  }
+  auto process_it = find_if(process_infos.begin(), process_infos.end(),
+                            [&pid](const ProcessInfo& info) { return info.pid() == pid; });
+  if (process_it == process_infos.end()) {
+    return nullptr;
+  }
+  LOG("Found process by pid, set target process");
+  std::shared_ptr<Process> process = std::make_shared<Process>();
+  process->SetID(process_it->pid());
+  process->SetName(process_it->name());
+  process->SetFullPath(process_it->full_path());
+  process->SetIs64Bit(process_it->is_64_bit());
+  LOG("Process info: pid:%d, name:%s, path:%s, is64:%d", process->GetId(), process->GetName(),
+      process->GetFullPath(), process->GetIs64Bit());
+  return process;
+}
+
+bool ClientGgp::LoadModuleAndSymbols() {
+  // Load modules for target_process_
+  ErrorMessageOr<std::vector<ModuleInfo>> result_module_infos =
+      process_client_->LoadModuleList(target_process_->GetId());
+  if (result_module_infos.has_error()) {
+    ERROR("Error retrieving modules: %s", result_module_infos.error().message());
+    return false;
+  }
+  // Find module that corresponds to the binary of target_process
+  const std::vector<ModuleInfo>& module_infos = result_module_infos.value();
+  LOG("List of modules");
+  for (const ModuleInfo& info : module_infos) {
+    LOG("name:%s, path:%s, size:%d, address_start:%d. address_end:%d, build_id:%s", info.name(),
+        info.file_path(), info.file_size(), info.address_start(), info.address_end(),
+        info.build_id());
+  }
+  std::string_view main_executable_name = target_process_->GetName();
+  auto module_it = find_if(module_infos.begin(), module_infos.end(),
+                           [&main_executable_name](const ModuleInfo& info) {
+                             return info.name() == main_executable_name;
+                           });
```

Comment:
You have to be careful here. A name of a process does not necessarily correspond to the name of the executable (elf file). The name can be set to something arbitrary. (It comes from `/proc/[pid]/comm` see here: https://man7.org/linux/man-pages/man5/proc.5.html)


 I would highly recommend using the path to match with modules instead of name.

Reply:
Oh, I see! Thanks for raising this, I will update it

Reply:
Done, thanks!